Sources/Controllers/2.0/OAMapSettingsContourLinesScreen.mm

```diff
@@ -365,28 +365,11 @@ - (void) generateData
 
 - (void) updateAvailableMaps
 {
-    CLLocation *loc = [[OARootViewController instance].mapPanel.mapViewController getMapLocation];
-    CLLocationCoordinate2D loca = loc.coordinate;
-    [OAResourcesUIHelper requestMapDownloadInfo:loca resourceType:OsmAnd::ResourcesManager::ResourceType::SrtmMapRegion onComplete:^(NSArray<OAResourceItem *>* res) {
-        @synchronized(_dataLock)
-        {
-            NSMutableArray<OARepositoryResourceItem *> *availableItems = [NSMutableArray array];
-            if (res.count > 0)
-            {
-                for (OAResourceItem * item in res)
-                {
-                    if ([item isKindOfClass:OARepositoryResourceItem.class])
-                    {
-                        OARepositoryResourceItem *resource = (OARepositoryResourceItem*)item;
-                        [availableItems addObject:resource];
-                    }
-                }
-                _mapItems = availableItems;
-            }
-            
-            [self generateData];
-            [tblView reloadData];
-        }
+    CLLocationCoordinate2D loca = [OAResourcesUIHelper getMapLocation];
+    [OAResourcesUIHelper getMapsForType:OsmAnd::ResourcesManager::ResourceType::SrtmMapRegion latLon:loca onComplete:^(NSArray<OARepositoryResourceItem *>* res) {
+        _mapItems = res;
```

Comment:
Why did you remove the ```@synchronized(_dataLock)``` block? Also, check if this code is always called from the main thread.